lib/egauge/client.rb

```diff
@@ -1,19 +1,29 @@
 require 'httpclient'
-require 'uri'
 
 module Egauge
   class Client
     include Egauge::PowerHelpers
 
+    EGAUGE_PATH = '/cgi-bin/egauge-show?'.freeze
+
     def initialize(base_url)
       @client = HTTPClient.new(base_url: base_url)
     end
 
-    def query(endpoint, options = {})
-      Egauge::Response.new(client.get(endpoint, options).body)
+    def query(options = {})
+      Egauge::Response.new(client.get(EGAUGE_PATH + query_string(options)).body)
     end
 
     private
       attr_reader :client
+
+      def query_string(options)
+        options.stringify_keys.map do |key, value|
+          query_parameter = String.new
+          query_parameter << key
```

Comment:
Can the above two lines be replaced with:
````ruby
query_parameter = key.dup
````

Reply:
It would need to be converted to a string first, but I think so yes.